SpatialGDK/Source/SpatialGDK/Private/Interop/AsyncPackageLoadFilter.cpp

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Interop/AsyncPackageLoadFilter.h"
+
+#include "UObject/UObjectGlobals.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+
+DEFINE_LOG_CATEGORY(LogAsyncPackageLoadFilter);
+
+namespace SpatialGDK
+{
+AsyncPackageLoadFilter::AsyncPackageLoadFilter(USpatialNetDriver* InNetDriver)
+	: NetDriver(InNetDriver)
+{
+}
+
+bool AsyncPackageLoadFilter::ProcessAndCheckEntityFilter(Worker_EntityId EntityId, const FString& ClassPath)
```

Comment:
Oh I did spend at least 10 mins trying to come up with a better name here. IsLoadedElseLoad could work. Will have a think.

My reasoning behind returning true was that this is a filter, and what it does is filter things. Did it filter it? Yes it did, return true. Did it not filter it and thus is no stopped, no? return false.

Reply:
on filters returning true- my mental model for filters was true means it passed the filter, rather than it was filtered. We could change everything to match the idea that true means filtered out, however that would be a bigger refactor, and I would rather for consistency reasons we went with what's already there for now if we don't do the refactor. "true means complete" is also documented in the subview documentation here https://improbableio.atlassian.net/wiki/spaces/GBU/pages/1842086036/Subviews

WDYT? Happy to revisit the boolean if it doesn't make sense

Reply:
Precedence is more than enough. Was just explaining my mental model. Will update 👍